watertap/unit_models/boron_removal.py

```diff
@@ -0,0 +1,1123 @@
+
+        # Add constraints for mass transfer terms
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            self.config.property_package.component_list,
+            doc="Mass transfer term",
+        )
+        def eq_mass_transfer_term(self, t, p, j):
+            if j == self.boron_name_id:
+                boron_out = pyunits.convert(
+                    self.mol_Boron[t],
+                    to_units=units_meta("amount") * units_meta("length") ** -3,
+                )
+                input_rate = self.control_volume.properties_in[t].flow_mol_phase_comp[
+                    p, self.boron_name_id
+                ]
+                exit_rate = (
+                    self.control_volume.properties_out[t].flow_vol_phase[p] * boron_out
+                )
+
+                loss_rate = input_rate - exit_rate
+                return self.control_volume.mass_transfer_term[t, p, j] == -loss_rate
+            elif j == self.borate_name_id:
+                borate_out = pyunits.convert(
+                    self.mol_Borate[t],
+                    to_units=units_meta("amount") * units_meta("length") ** -3,
+                )
+                input_rate = self.control_volume.properties_in[t].flow_mol_phase_comp[
+                    p, self.borate_name_id
+                ]
+                exit_rate = (
+                    self.control_volume.properties_out[t].flow_vol_phase[p] * borate_out
+                )
+
+                loss_rate = input_rate - exit_rate
+                return self.control_volume.mass_transfer_term[t, p, j] == -loss_rate
+            elif j == self.proton_name_id:
+                p_out = pyunits.convert(
+                    self.mol_H[t],
+                    to_units=units_meta("amount") * units_meta("length") ** -3,
+                )
+                input_rate = self.control_volume.properties_in[t].flow_mol_phase_comp[
+                    p, self.proton_name_id
+                ]
+                exit_rate = (
+                    self.control_volume.properties_out[t].flow_vol_phase[p] * p_out
+                )
+
+                loss_rate = input_rate - exit_rate
+                return self.control_volume.mass_transfer_term[t, p, j] == -loss_rate
+            elif j == self.hydroxide_name_id:
+                h_out = pyunits.convert(
+                    self.mol_OH[t],
+                    to_units=units_meta("amount") * units_meta("length") ** -3,
+                )
+                input_rate = self.control_volume.properties_in[t].flow_mol_phase_comp[
+                    p, self.hydroxide_name_id
+                ]
+                exit_rate = (
+                    self.control_volume.properties_out[t].flow_vol_phase[p] * h_out
+                )
+
+                loss_rate = input_rate - exit_rate
+                return self.control_volume.mass_transfer_term[t, p, j] == -loss_rate

```

Comment:
I skimmed this quickly so may be wrong, but can these be condensed? It looks like you might only need the conditional to decide which component molarity will be converted.

e.g.,: 
```
if j == self.proton_name_id:
    molarity = self.mol_H[t]
elif j == self.hydroxide_name_id:
   molarity = self.mol_OH[t]
.....
```

Reply:
Wouldn't change functionality but would reduce lines of code

Reply:
Yes, I think that may work for most of them.